modin/backends/pandas/query_compiler.py

```diff
@@ -1905,12 +1905,64 @@ def mode_builder(df):
         return self.__constructor__(new_modin_frame).dropna(axis=axis, how="all")
 
     def fillna(self, **kwargs):
+        squeeze_self = kwargs.pop("squeeze_self", False)
+        squeeze_value = kwargs.pop("squeeze_value", False)
         axis = kwargs.get("axis", 0)
         value = kwargs.get("value")
         method = kwargs.get("method", None)
         limit = kwargs.get("limit", None)
         full_axis = method is not None or limit is not None
-        if isinstance(value, dict):
+        if isinstance(value, BaseQueryCompiler):
+            kwargs.pop("value")
```

Comment:
Why do we need to pop `value`?

Reply:
Because `value` is passed separately and if `pop` is not done, call produces `TypeError: fillna() got multiple values for keyword argument 'value'`.

Reply:
why don't you just do `value = kwargs.pop("value")` above?

Reply:
Ok.